### PR TITLE
Api 34935 bgs person create

### PR DESCRIPTION
**Pull request by acovrig**
## Summary
- Adds a connector to BGS's `VnpPersonCreate` endpoint.

## Related issue(s)
- [API-34935](https://jira.devops.va.gov/browse/API-34935)

## Testing done
- [x] *New code is covered by unit tests*
- How I tested:
  Added this to `ClaimsApi::V2::Veterans::PowerOfAttorney::BaseController`:
  ```
  require 'bgs_service/vnp_person_service'
  require 'bgs_service/vnp_proc_service_v2'
  require 'bgs_service/vnp_ptcpnt_service'
  ```
  and in the `#show` method:
  ```
  vnp_proc_id = VnpProcServiceV2.new(
    external_key: target_veteran.participant_id,
    external_uid: target_veteran.participant_id
  ).vnp_proc_create[:vnp_proc_id]
  vnp_ptcpnt_id = VnpPtcpntService.new(
    external_key: target_veteran.participant_id,
    external_uid: target_veteran.participant_id
  ).vnp_ptcpnt_create(
    vnp_proc_id:,
    jrn_dt: Time.current.iso8601,
    jrn_lctn_id: Settings.bgs.client_station_id,
    jrn_obj_id: Settings.bgs.application,
    jrn_status_type_cd: 'U',
    jrn_user_id: Settings.bgs.client_username,
    ptcpnt_type_nm: 'Person'
  )[:vnp_ptcpnt_id]
  data = {
    vnp_proc_id:,
    vnp_ptcpnt_id:,
    first_nm: target_veteran.first_name,
    last_nm: target_veteran.last_name
  }
  res = VnpPersonService.new(
    external_key: target_veteran.participant_id,
    external_uid: target_veteran.participant_id
  ).vnp_person_create(data)
  # ).vnp_person_create(data, target_veteran:)
  # ).vnp_person_create(data, icn: params[:veteranId])
  return render json: { data: res }
  ```
  swapping out the last 3 lines to try the different signatures of the method. Then I was able to GET `/claims/v2/veterans/:icn/power-of-attorney` to see the response

## What areas of the site does it impact?
Will impact v2; no impact yet

## Acceptance criteria
- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback
- Is the `icn:` and/or `target_veteran` optional params useful or not?
  The `participant_id` seems to be one from the `vnp_ptcpnt_create` call, so IDK if the `target_veteran` or MPI response would have a usable `participant_id` anywho?
- Is it OK to do `given_names.first` to get the `first_name`?

### CODE REVIEW

**Comment by stevenjcumming** [comment]
@acovrig Could you get someone on your team to review and approve? Then I'll review it.

**Review by stiehlrod — COMMENTED** [review]
Successful test
Screenshot 2024-04-02 at 1 11 16 PM

**Review by stiehlrod — COMMENTED** [review]
-> Is the icn: and/or target_veteran optional params useful or not?
The participant_id seems to be one from the vnp_ptcpnt_create call, so IDK if the target_veteran or MPI response would have a usable participant_id anywho?

I don't think we need to pass in the icn or target_veteran.

File: modules/claims_api/lib/bgs_service/vnp_person_service.rb [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    def jrn
+      {
+        jrnDt: Time.current.iso8601,
+        jrnLctnId: Settings.bgs.client_station_id,
+        jrnStatusTypeCd: 'U',
+        jrnUserId: Settings.bgs.client_username,
+        jrnObjId: Settings.bgs.application
+      }
+    end
+
```
Comment (stiehlrod) [inline modules/claims_api/lib/bgs_service/vnp_person_service.rb:52]: @acovrig I don't think we need to pass in the icn or target_veteran if we are already passing `vnpPtcpntId: target_veteran.participant_id,` as an arg.

**Comment by stevenjcumming** [comment]
@nihil2501 please ask a teammate for an approval, then I will review it

**Review by FonzMP — APPROVED** [review]
:shipit:

**Pull request merged**